compiler/riscv/codegen/RVSystemLinkage.cpp

```diff
@@ -224,39 +167,17 @@ void
 TR::RVSystemLinkage::initRVRealRegisterLinkage()
    {
    TR::Machine *machine = cg()->machine();
-   TR::RealRegister *reg;
-   int icount;
-
-   reg = machine->getRealRegister(TR::RealRegister::RegNum::zero);
-   reg->setState(TR::RealRegister::Locked);
-   reg->setAssignedRegister(reg);
-
-   reg = machine->getRealRegister(TR::RealRegister::RegNum::ra);
-   reg->setState(TR::RealRegister::Locked);
-   reg->setAssignedRegister(reg);
-
-   reg = machine->getRealRegister(TR::RealRegister::RegNum::sp);
-   reg->setState(TR::RealRegister::Locked);
-   reg->setAssignedRegister(reg);
-
-   reg = machine->getRealRegister(TR::RealRegister::RegNum::gp);
-   reg->setState(TR::RealRegister::Locked);
-   reg->setAssignedRegister(reg);
-
-   reg = machine->getRealRegister(TR::RealRegister::RegNum::tp);
-   reg->setState(TR::RealRegister::Locked);
-   reg->setAssignedRegister(reg);
-
-   reg = machine->getRealRegister(TR::RealRegister::RegNum::s0); // FP
-   reg->setState(TR::RealRegister::Locked);
-   reg->setAssignedRegister(reg);
 
+   FOR_EACH_RESERVED_REGISTER(machine, _properties,
+         reg->setState(TR::RealRegister::Locked);
+         reg->setAssignedRegister(reg);
+   );
```

Comment:
Personally I don't find this macro useful. To me it adds an extra layer of abstraction/complexity when it is merely just a for loop. I recall when I first started with the codebase and didn't have a proper editor with symbols setup and I saw the:

https://github.com/eclipse/omr/blob/eb4f5a875f3b30b1dd3512f77b3c7a1293a78724/compiler/env/OMRCPU.cpp#L150-L160

`OMRPORT_ACCESS_FROM_OMRPORT` and similar macros. I could never find what that was doing or where this magical `privatePortLibrary` local variable was coming from. It turns out all that macro does is declare a local variable:

https://github.com/eclipse/omr/blob/eb4f5a875f3b30b1dd3512f77b3c7a1293a78724/include_core/omrport.h#L70

And then `omrsysinfo_*` macros consume that local variable silently:

https://github.com/eclipse/omr/blob/eb4f5a875f3b30b1dd3512f77b3c7a1293a78724/include_core/omrport.h#L2688

This is super unintuitive to me as a new contributor to the codebase. To me simply declaring the local variable and using the API by passing all arguments is much more clear. No need to peel layers of complexity to understand what is happening, i.e. "in the `FOR_EACH_RESERVED_REGISTER` which argument needs a comma, and which argument needs a semicolon?".

I'd say a similar principle applies here.

Reply:
> Personally I don't find this macro useful.

I must say I find it useful in a sense it makes it much shorter so the body is not "burried" in copy-pasted boilerplate code (*) - it is used on several places in linkage code. 

(*) like geting a `TR::RealRegister*` for given `TR::RealRegister::RegNum` while filtering by register flags. 

I do share your dislike for macros (and `#define` of constants) and I'd like to replace this macro with range-based for-loop so one can do something like 

    for (auto reg : this->getReservedRegisters()) { ... }

but as of now, coding standards forbid range-based for-loops (but they're out of date, AFAIK, see #5592). Also for time reasons, I left this exercise for a future.